p2p/net/mock/mock_net.go

```diff
@@ -85,6 +85,7 @@ func (mn *mocknet) AddPeerWithPeerstore(p peer.ID, ps pstore.Peerstore) (host.Ho
 	}
 
 	h := bhost.New(n)
+	h.NegotiateTimeout = 0
```

Comment:
Disabled for tests because the underlying `net.Conn` pipe that mocknet uses doesnt support `SetDeadline` :/